mesheryctl.sh

```diff
@@ -1,16 +1,23 @@
 #!/usr/bin/env bash
 
+set -o errexit
+set -o nounset
+set -o pipefail
+
 SCRIPT_DIR=$(dirname -- "$(readlink -f "${BASH_SOURCE[0]}" || realpath "${BASH_SOURCE[0]}")")
 
 declare -A adapters
 adapters["istio"]=meshery-istio:10000
 adapters["linkerd"]=meshery-linkerd:10001
 adapters["consul"]=meshery-consul:10002
 adapters["octarine"]=meshery-octarine:10003
+adapters["nsm"]=meshery-nsm:10004
```

Comment:
Why are we not using `mesheryctl system context create <context-name> --adapters meshery-osm`?

Reply:
@leecalcote this key-value map is actually used for docker networking and for specifying adapter in `mesh deploy` command. :sweat_smile:

Reply:
Also, since we have the [service_mesh](https://github.com/layer5io/meshery-smp-action/blob/89cde7517e4f624da584c5f45ce949503c48810d/action.yml#L41) field in action configuration now, I think it should be possible to have a single adapter deployed now. But wouldn't this lead to bad UX as performance profiles might have a different service mesh specified in them? // @navendu-pottekkat

Reply:
The `service_mesh` field is only for overriding the service mesh specified in the performance configuration file right? So, whatever service mesh they are testing on, will have its adapter running and the service mesh name will be part of the performance test configuration. How would this be a bad UX?

Reply:
Hmm... So I thought `service_mesh` would mostly be used by `mesh deploy` and if the specified test configuration had a different service mesh, this will conflict. But I missed the part around overriding. Got your point :+1: